system/lib/wasmfs/syscalls.cpp

```diff
@@ -492,7 +492,14 @@ static __wasi_fd_t doOpen(path::ParsedParent parsed,
   if (flags & O_TRUNC) {
     if (child->is<DataFile>()) {
       if (fileMode & WASMFS_PERM_WRITE) {
-        child->cast<DataFile>()->locked().setSize(0);
+        // TODO: Linux still truncates the file in when it's opened only for
+        // reading, so we should too. But until we get proper error reporting
+        // from setSize (which we can then ignore here), the OPFS backend will
+        // crash in this case because it does not support truncating files open
+        // only for reading.
+        if (accessMode != O_RDONLY) {
+          child->cast<DataFile>()->locked().setSize(0);
```

Comment:
Could we put this workaround in the OPFS code and not in the generic syscall code? That is, OPFS setSize could handle it.

Reply:
That would require `setSize` to fail silently for now, but I would rather have this temporary hack and have `setSize` fail loudly for now than remove this hack and have `setSize` fail silently.

Reply:
Oh, because `setSize` has no return value of an error code? That sounds good for now then with a TODO

Reply:
Yeah, exactly. Is there any extra information you would want to see beyond the current TODO?

Reply:
I'm still not clear about what you'd like to see. There is already a TODO, but would you like to see an additional or modified TODO?

Reply:
Sorry - I meant lgtm as it is,